src/pages/workspace/WorkspaceInitialPage.tsx

```diff
@@ -347,6 +349,24 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, reimbursementAcc
         PolicyUtils.goBackFromInvalidPolicy();
     }, [policy, prevPolicy]);
 
+    // We are checking if the user can access the route.
+    // If user can't access the route, we are dismissing any modals that are open when the NotFound view is shown
+    const canAccessRoute = activeRoute && menuItems.some((item) => item.routeName === activeRoute);
+
+    useEffect(() => {
+        if (!shouldShowNotFoundPage && canAccessRoute) {
+            return;
+        }
+        if (CONFIG.USE_REACT_STRICT_MODE && wasRendered.current) {
```

Comment:
I don't like using `USE_REACT_STRICT_MODE` flag here. It's not used anywhere yet except in `StrictModeWrapper`.

Reply:
@situchan it wasn't necessary in the end, so we've removed it.